osvr/RenderKit/RenderManagerD3D11ATW.h

```diff
@@ -528,62 +514,68 @@ namespace osvr {
+                    ID3D11Texture2D *texture2D = nullptr;
+                    IDXGIResource* dxgiResource = nullptr;
+                    hr = textureToMap->QueryInterface(__uuidof(IDXGIResource), (LPVOID*)&dxgiResource);
+                    if (FAILED(hr)) {
+                      m_log->error()
+                        << "RenderManagerD3D11ATW::"
+                        << "RegisterRenderBuffersInternal: Can't get the IDXGIResource for the texture resource.";
+                      setDoingOkay(false);
+                      return false;
+                    }
+
+                    // now get the shared HANDLE
+                    hr = dxgiResource->GetSharedHandle(&newInfo.sharedResourceHandle);
+                    if (FAILED(hr)) {
+                      m_log->error()
+                        << "RenderManagerD3D11ATW::"
+                        << "RegisterRenderBuffersInternal: Can't get the shared handle from the dxgiResource.";
+                      setDoingOkay(false);
+                      return false;
+                    }
+                    dxgiResource->Release(); // we don't need this anymore
+
+                                             // Get a pointer on our device to the texture we're getting
+                                             // from the caller's device.
+                    hr = atwDevice->OpenSharedResource(newInfo.sharedResourceHandle, __uuidof(ID3D11Texture2D),
+                      (LPVOID*)&texture2D);
+                    if (FAILED(hr)) {
+                      m_log->error() << "RenderManagerD3D11ATW::"
+                        << "RegisterRenderBuffersInternal: - failed to open shared resource "
+                        << "for buffer " << i;
+                      setDoingOkay(false);
+                      return false;
+                    }
 
                     // We do not need a render target view for the ATW thread -- it will
                     // only be reading from the buffer, not rendering into it.  Our base
                     // class will create our RenderTargetView the first time the app calls
                     // Render().
+                    newInfo.atwBuffer.D3D11 = new osvr::renderkit::RenderBufferD3D11();
+                    newInfo.atwBuffer.D3D11->colorBuffer = texture2D;
                     newInfo.atwBuffer.D3D11->colorBufferView = nullptr; // We don't need this.
-
-					newInfo.atwBuffer.D3D11->depthStencilBuffer = nullptr;
-					newInfo.atwBuffer.D3D11->depthStencilView = nullptr;
-
+		    newInfo.atwBuffer.D3D11->depthStencilBuffer = nullptr;
+		    newInfo.atwBuffer.D3D11->depthStencilView = nullptr;
                     renderBuffers.push_back(newInfo.atwBuffer);
                   }
 
+                  // Lock our mutex so that we're not rendering while new buffers are
+                  // being added or old ones modified.
+                  std::lock_guard<std::mutex> lock(mLock);
```

Comment:
Interesting.  The standard defines that the guard begins at the time of construction of the lock_guard object, which is when it is reached in the code and lasts until that block is exited, so this should present anything funny.  It could help clarify things for someone looking at the code, so long as it didn't confuse them more because of a close block not associated with a statement.

This beginning-of-block placement was not done throughout the original code before the changes, but I've gone ahead and pushed a commit that does this.